accepted/2020/net5/net5.md

```diff
@@ -0,0 +1,854 @@
+might not make sense to define `net5.0-wasm`. Rather, it would make more sense to
+define `net5.0-browser`. The rationale is that the browser will run WASM in a
+sandboxed environment which equates to having different native API sets.
+
+Any host that controls the JS runtime (for example, Node.js) could decide to expose
+different/less constrained OS, which might give rise to other TFMs, such as
+`net5.0-node`.
+
+### What about native dependencies?
+
+We don't plan to support varying API surface based on runtime characteristics
+(x86/x64, AOT/JIT etc). This will continue to be supported via
+the `runtime/<RID>` folder.
+
+### Will the new TFMs simplify the project files too?
+
+[@cartermp](https://github.com/cartermp) asked:
+
+> Does specifying `net5.0-windows` obviate the current three things you need to
+> specify?
+>
+> * netcoreapp3.x
+> * Desktop SDK attribute
+> * UseWindowsForms/UseWPF
+>
+> IOW are we going to have to support both formats moving forward? What about
+> converting apps?
+
+Generally, it does not. The idea I've heard is that all project types will be
+unified to use `Sdk="Microsoft.NET.Sdk"` in order to make multi-targeting
+easier. Customizations (for example, specific targets and references) would be brought
+in via `UseXxx` properties, akin to how Windows Forms and WPF work in .NET Core
+today. The reason is that in many cases the TFM alone isn't specific enough to
+decide what kind of app are you building:
+
+* `net5.0`. Is a class library/console app, an ASP.NET Core app, or a Blazor app?
+* `net5.0-windows`. Are you building a Windows Forms app or a WPF app? Are you
+  using both Windows Forms and WPF or just one?
+
+The nice thing about properties is that they naturally compose. If certain
+combinations aren't possible, they can be blocked relatively easily.
+
+However, at this point it's still unclear whether the SDK unification will work
+this way. One concern was that SDKs also bring in new item groups and might have
+conflicting defaults for properties; this works today because the SDK can bring
+in .props before the project file. When we rely on properties in the project
+file, we need to bring those in the .targets (that is, the bottom of the project
+file). While not impossible, this might force us to have knowledge in the base
+SDK that can't be easily extended via optional components.
+[@mhutch](https://github.com/mhutch) is working on a document specifically
+around SDK convergence.
+
+### Why is there no TFM for Linux?
+
+The primary reason for OS specific TFMs is to vary API surface, not for varying
+behavior. RIDs allow varying behavior and have support for various Linux
+flavors. Specifically, TFMs aren't (primarily) meant to allow calling P/Invokes
+under `#if`, most of the time that should be done by doing runtime checks or by
+using RIDs. The primary reason for a TFM is to exclude large amounts of managed
+representations for OS technologies (WinForms, WPF, Apple's NS APIs, Android
+etc).
```

Comment:
I have a question about this. Doesn't .NET 5 support [mono WinForms](https://www.mono-project.com/docs/gui/winforms/) implementation or [GTK#](https://www.mono-project.com/docs/gui/gtksharp/)?

Reply:
No. .NET 5's WinForms is the open source & ported version of .NET Framework's WinForms implementation and is Windows only. We have no plans to make it cross-platform. This decision was based on input from the Mono folks who wrote the GTK implementation. We want to tackle cross-platform UI, but not by porting a Windows-only design because it's never going to be great and will always remain a very leaky abstraction.

Reply:
So, it means WinForms app running on Mono can't be ported to .NET 5 and have to keep using Mono.
Okay, Thank you for your answer.